include/exadg/postprocessor/pointwise_output_generator_base.cpp

```diff
@@ -0,0 +1,291 @@
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#include <exadg/postprocessor/pointwise_output_generator_base.h>
+
+#include <deal.II/lac/full_matrix.h>
+
+#include <exadg/utilities/create_directories.h>
+#include <exadg/utilities/print_functions.h>
+
+namespace ExaDG
+{
+template<int dim>
+PointwiseOutputDataBase<dim>::PointwiseOutputDataBase()
+  : write_output(false),
+    directory("output/"),
+    filename("name"),
+    update_points_before_evaluation(false),
+    start_time(std::numeric_limits<double>::max()),
+    end_time(std::numeric_limits<double>::max()),
+    interval_time(std::numeric_limits<double>::max())
+{
+}
+
+template<int dim>
+void
+PointwiseOutputDataBase<dim>::print(dealii::ConditionalOStream & pcout) const
+{
+  ExaDG::print_parameter(pcout, "PointwiseOutput", write_output && evaluation_points.size() > 0);
+
+  if(write_output && evaluation_points.size() > 0)
+  {
+    ExaDG::print_parameter(pcout, "Output directory", directory);
+    ExaDG::print_parameter(pcout, "Name of output file", filename);
+    ExaDG::print_parameter(pcout, "Output start time", start_time);
+    ExaDG::print_parameter(pcout, "Output end time", end_time);
+    ExaDG::print_parameter(pcout, "Output interval time", interval_time);
+    ExaDG::print_parameter(pcout,
+                           "Update Points before evaluation",
+                           update_points_before_evaluation);
+  }
+}
+
+template class PointwiseOutputDataBase<2>;
+template class PointwiseOutputDataBase<3>;
+
+template<int dim, typename Number>
+PointwiseOutputGeneratorBase<dim, Number>::PointwiseOutputGeneratorBase(MPI_Comm const & comm)
+  : mpi_comm(comm), counter(0), reset_counter(true)
+{
+}
+
+template<int dim, typename Number>
+void
+PointwiseOutputGeneratorBase<dim, Number>::setup_base(
+  dealii::DoFHandler<dim> const &      dof_handler_in,
+  dealii::Mapping<dim> const &         mapping_in,
+  PointwiseOutputDataBase<dim> const & pointwise_output_data_in)
+{
+  pointwise_output_data = pointwise_output_data_in;
+
+  if(pointwise_output_data.write_output && pointwise_output_data.evaluation_points.size() > 0)
+  {
+    dof_handler = &dof_handler_in;
+    mapping     = &mapping_in;
+
+    // allocate memory for hyperslap
```

Comment:
spelling: hyperslap vs. hyperslab?

Reply:
resolved in [6e0c9b3](https://github.com/exadg/exadg/pull/255/commits/6e0c9b3db6fa171da3b797e3347ce1c0720106a2)